homeassistant/components/ecobee.py

```diff
@@ -15,12 +16,19 @@
 
 DOMAIN = "ecobee"
 NETWORK = None
-HOLD_TEMP = 'hold_temp'
+CONF_HOLD_TEMP = 'hold_temp'
 
 REQUIREMENTS = [
     'https://github.com/nkgilley/python-ecobee-api/archive/'
     '4856a704670c53afe1882178a89c209b5f98533d.zip#python-ecobee==0.0.6']
 
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_API_KEY): str,
```

Comment:
It is required the first time you run the ecobee component, but after that the api key is saved to ecobee.conf and can be removed from the configuration.yaml.  I can change to required if you guys still think that's the best way to go.